packages/venia-concept/src/components/MiniCart/product.js

```diff
@@ -1,161 +1,116 @@
-                            {label} : {value}
-                        </dt>
-                    </Fragment>
-                ))}
-            </dl>
-        ) : null;
-    }
+    useEffect(() => {
+        // After the imageRef mounts, get its --image-width so
+        // we can pass it to `resourceUrl`.
+        const imageWidth = getComputedStyle(imageRef.current).getPropertyValue(
+            '--image-width'
+        );
 
-    get mask() {
-        const { classes } = this.props;
-        return this.state.isLoading ? <div className={classes.mask} /> : null;
-    }
+        if (imageWidth) {
+            // parseFloat to strip off any units like 'px'.
+            const parsedImageWidth = parseFloat(imageWidth);
+            setBackgroundImage(
+                `url(${resourceUrl(image.file, {
+                    type: 'image-product',
+                    width: parsedImageWidth
+                })})`
+            );
+        }
+    }, [imageRef]);
 
-    styleImage(image) {
-        return {
-            minHeight: imageHeight, // min-height instead of height so image will always align with grid bottom
-            width: imageWidth,
-            backgroundImage: `url(${resourceUrl(image.file, {
-                type: 'image-product',
-                width: imageWidth
-            })})`
-        };
-    }
+    const handleFavoriteItem = useCallback(() => {
+        setIsFavorite(!isFavorite);
+    }, [isFavorite, setIsFavorite]);
+    const handleEditItem = useCallback(() => {
+        beginEditItem(item);
+    }, [beginEditItem, item]);
+    const handleRemoveItem = useCallback(() => {
+        setIsLoading(true);
 
-    render() {
-        const { options, props, mask } = this;
-        const { classes, item, currencyCode } = props;
-        const favoritesFill = { fill: 'rgb(var(--venia-teal))' };
+        // TODO: prompt user to confirm this action?
+        removeItemFromCart({ item });
+    }, [item, removeItemFromCart, setIsLoading]);
 
-        return (
-            <li className={classes.root}>
-                <div
-                    className={classes.image}
-                    style={this.styleImage(item.image)}
-                />
-                <div className={classes.name}>{item.name}</div>
-                {options}
-                <div className={classes.quantity}>
-                    <div className={classes.quantityRow}>
-                        <span>{item.qty}</span>
-                        <span className={classes.quantityOperator}>{'×'}</span>
-                        <span className={classes.price}>
-                            <Price
-                                currencyCode={currencyCode}
-                                value={item.price}
-                            />
-                        </span>
-                    </div>
+    return (
+        <li className={classes.root}>
+            <div
+                className={classes.image}
+                ref={imageRef}
```

Comment:
Switched to using an `img` instead in https://github.com/magento-research/pwa-studio/pull/1266/commits/5147244ff5b17e4ab8fdcf299f7f804eb926d831.

Yeah, my objective here was to set height and width in CSS instead of the JS. I think mine is a personal preference so if you feel strongly about it I can move it back.

Reply:
I think we should use the semantics of HTML when possible, so yea in this case I would prefer this be an `<img>`

Reply:
See https://gist.github.com/paulirish/5d52fb081b3570c81e3a

Reply:
Resolved by https://github.com/magento-research/pwa-studio/pull/1266/commits/2826d2c12efd832d7b85e75eb75687677f0b61b0.